test/alternator/test_gsi.py

```diff
@@ -163,9 +163,25 @@ def test_gsi_strong_consistency(test_table_gsi_1):
 
 # Test that setting an indexed string column to an empty string is illegal,
 # since keys cannot contain empty strings
-def test_gsi_empty_value(test_table_gsi_2):
+def test_gsi_empty_value(test_table_gsi_3):
     with pytest.raises(ClientError, match='ValidationException.*empty'):
-        test_table_gsi_2.put_item(Item={'p': random_string(), 'x': ''})
+        test_table_gsi_3.put_item(Item={'p': random_string(), 'a': ''})
+    with pytest.raises(ClientError, match='ValidationException.*empty'):
+        test_table_gsi_3.put_item(Item={'p': random_string(), 'b': ''})
```

Comment:
Can you please keep both tests, one for test_table_gsi_2 and one for test_table_gsi_3? I think it makes sense to have both cases, which probably reach slightly different code paths.

Reply:
ok

Reply:
following the same logic I'll also split `test_gsi_null_value`